### PR TITLE
Spark: Moves Distribution and Ordering Utils to Spark3 Module

### PR DESCRIPTION
**Pull request by RussellSpitzer**
To enable core Spark3 code to use distribution and ordering information when
performing writes or other operations. One example is to allow Sort Strategy
to transform SortOrder information so it can be used when rewriting data files.

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
@aokolnychyi + @rdblue I would like to have access to these Utilities for doing some work with the Sort Rewrite Action. If anyone has another Idea on how to get access to this functionality I would be glad to hear it. Ideally I would like to keep Scala and this code out of the Spark3 module, but I can't have an Action use code from within the Extensions module.

File: build.gradle [hunk 1]
```diff
@@ -986,7 +986,14 @@ if (jdkVersion == '8') {
...
 project(':iceberg-spark3') {
+  apply plugin: 'scala'
+
   sourceSets {
+    main {
+      // Use Scala Compiler for everything so we can do cross compilation within the module
+      scala.srcDirs = ['src/main/scala', 'src/main/java']
+      java.srcDirs = []
+    }
     // Compile test source against Spark 3.1 and main classes compiled against Spark 3.0
```
Comment (wypoon) [inline build.gradle:997]: Do we need to do anything with the spark31 sourceSet?

File: build.gradle [hunk 2]
```diff
@@ -986,7 +986,14 @@ if (jdkVersion == '8') {
 }
 
 project(':iceberg-spark3') {
+  apply plugin: 'scala'
+
   sourceSets {
+    main {
+      // Use Scala Compiler for everything so we can do cross compilation within the module
```
Comment (rdblue) [inline build.gradle:993]: I don't think I understand why we can't use the default scala plugin settings. This is disabling Java compilation by removing srcDirs. Should we just remove the Java plugin?

**Comment by rdblue** [comment]
I tried to convert some of these classes to Java today and didn't get anywhere. Some of the traits we need to implement just don't exist for some reason and I think it's probably better just to do it this way.

**Comment by RussellSpitzer** [comment]
@rdblue Are we good to go with this?

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Looks good to me. The other build settings weren't needed after all?

**Comment by RussellSpitzer** [comment]
> Looks good to me. The other build settings weren't needed after all?

Yep I'm not sure what I was thinking :)

File: build.gradle (hunk 1 above)
Comment (RussellSpitzer) [reply build.gradle:992]: Nope, we still use the "main" source set

**Comment by RussellSpitzer** [comment]
Thanks for the review @wypoon + @rdblue

**Pull request merged**